crates/gitbutler-tauri/src/events.rs

```diff
@@ -101,22 +76,22 @@ impl Event {
                 "filePath": file_path,
                 "contents": contents,
             }),
-            project_id: *project_id,
+            project_id,
         }
     }
 
-    pub fn session(project_id: &ProjectId, session: &sessions::Session) -> Self {
+    pub fn session(project_id: ProjectId, session: &sessions::Session) -> Self {
         Event {
             name: format!("project://{}/sessions", project_id),
             payload: serde_json::to_value(session).unwrap(),
-            project_id: *project_id,
+            project_id,
         }
     }
 
     pub fn deltas(
-        project_id: &ProjectId,
-        session_id: &SessionId,
-        deltas: &Vec<deltas::Delta>,
+        project_id: ProjectId,
+        session_id: SessionId,
+        deltas: &[deltas::Delta],
```

Comment:
Just a thought: we could also use a generic here for `AsRef<[...]>`.

Reply:
Absolutely - I usually avoid generics until tests tell me they want it to make it more pleasant to work with the API under test.